lms/djangoapps/instructor/management/commands/dump_userinfo.py

```diff
@@ -231,18 +263,41 @@ def handle(self, *args, **options):
         intervals = int(0.10 * total)
         if intervals > 100 and verbose:
             intervals = 101
+        if intervals < 1:
+            intervals = 1
 
-        sys.stdout.write("Processing users")
+        sys.stdout.write("Processing users \n")
 
+        user_scores = []
         for profile in profiles:
             user_id = profile['user__id']
+            user = User.objects.get(id=user_id)
             self.print_progress(count, intervals, verbose)
 
+            # Create dict which will pass to the rows CSV
             student_dict = {
                 'Credits Issued': 0.0,  # XXX should be revisited when credit count functionality implemented
                 'Attested': False,
             }
 
+            if include_grades:
+                # Get scores for each exam and user in a course,
+                # then create a JSON that will associate the scores with the user
+                user_json = {'name':user}
+                scores = self.get_scores(user, course_id)
+                exams = []
+                for grade in scores['section_breakdown']:
+                    exam_scores = {'score':grade['percent'],'label':grade['label']}
+                    exams.append(exam_scores)
+                user_json['exams'] = exams
+                user_scores.append(user_json)
+                for user in user_scores:
```

Comment:
I'm a bit concerned/confused with this loop,
specifically that it's nested inside the `for profile in profiles:` loop.

`user_scores` will continue to grow upon each iteration of the profiles list.
So then here, we'll be iterating over more and more users each time,
when it seems like we probably only care about the most recent one, right?

Can we remove this outer `for user in user_scores:` loop and just say:

```python
for exam in user_json['exams']:
    label = exam['label']
    student_dict[label] = exam['score']
```

This would appear to reduce many unnecessary iterations.
Or am I missing something?

Reply:
`user_scores` was created to store the scores in an independent way, but your solution it's cleaner. Thanks for the observation